src/openforms/forms/api/serializers/form_variable.py

```diff
@@ -177,18 +180,47 @@ def validate(self, attrs):
                     }
                 )
 
-        # prefill plugin and attribute must both or both not be set
+        # Check the combination of the provided prefill-attributes (see the model constraints)
+        source = attrs.get("source") or ""
         prefill_plugin = attrs.get("prefill_plugin") or ""
         prefill_attribute = attrs.get("prefill_attribute") or ""
-        if (prefill_plugin and not prefill_attribute) or (
-            not prefill_plugin and prefill_attribute
-        ):
+        prefill_options = attrs.get("prefill_options")
+
+        if prefill_plugin and prefill_options and prefill_attribute:
             raise ValidationError(
                 {
-                    "prefill_attribute": _(
-                        "Prefill plugin and attribute must both be specified."
+                    "prefill_attribute_options": _(
```

Comment:
```suggestion
                    "prefill_attribute": _(
```
either attach it to one particular field, or just `raise ValidationError(message)` to make it a non-field error